designs/lists-design.md

```diff
@@ -0,0 +1,175 @@
+## Test cases for lists
+
+
+### 1. Create a list with some elements in it
+
+*Input:*
+```
+a: [int] = None
+a = [1, 2, 3]
+```
+*Output:*
+(no output)
+
+---
+
+### 2. Create a list with no elements in it
+
+*Input:*
+```
+a: [int] = None
+a = []
+```
+*Output:*
+(no output)
+
+---
+
+### 3. Access an element in the list
+
+*Input:*
+```
+a: [int] = None
+a = [2, 4, 6, 8]
+a[0]
+```
+*Output:*
+```
+2
+```
+
+---
+
+### 4. Access an element, out of bounds
+*Input:*
+```
+a: [int] = None
+a = [2, 4, 6, 8]
+a[4]
+```
+*Output:*
+```
+Error: Index 4 out of bounds
```

Comment:
When does this error get thrown?

Reply:
We haven't determined this yet

Reply:
We're trying to throw a runtime error, which we will need to create.